lib/sensors/i2c_mux.hpp

```diff
@@ -68,68 +72,61 @@ std::optional<std::array<T, N>> Mux<T, N>::readAllChannels()
     // First ensure correct channel is selected
     const auto channel_select_result = selectChannel(channel);
     if (channel_select_result == core::Result::kFailure) {
-      logger_.log(core::LogLevel::kFatal, "Failed to select mux channel %d", channel);
+      logger_.log(core::LogLevel::kFatal, "Failed to select channel %d from the i2c mux", channel);
       return std::nullopt;
     }
-    // Next ensure sensor is configured for operation
-    const auto configure_result = sensor->configure();
-    if (configure_result == core::Result::kFailure) {
-      logger_.log(core::LogLevel::kFatal, "Failed to configure sensor at mux channel %d", channel);
-      ++num_unusable_sensors;
-      continue;
-    }
-    // Finally read sensor data
+    // Then read sensor data
     const auto sensor_data = sensor->read();
     if (!sensor_data) {
-      logger_.log(core::LogLevel::kFatal, "Failed to get mux data from channel %d", channel);
+      logger_.log(core::LogLevel::kFatal, "Failed to get i2c mux data from channel %d", channel);
       ++num_unusable_sensors;
       continue;
     }
     mux_data.at(i)                    = sensor_data.value();
     const auto closing_channel_result = closeAllChannels();
     if (closing_channel_result == core::Result::kFailure) {
-      logger_.log(core::LogLevel::kFatal, "Failed to close all mux channels while reading");
+      logger_.log(core::LogLevel::kFatal, "Failed to close all i2c mux channels while reading");
       return std::nullopt;
     }
   }
   // If too many sensors are unusable, assume mux is faulty
   if (num_unusable_sensors > max_num_unusable_sensors_) {
     logger_.log(core::LogLevel::kFatal,
-                "Failed to read from more than %0.f%% of sensors on the mux",
+                "Failed to read from more than %0.f%% of sensors on the i2c mux",
                 kFailureThreshold * 100);
     return std::nullopt;
   }
   return mux_data;
 }
 
 template<typename T, std::uint8_t N>
-core::Result Mux<T, N>::selectChannel(const std::uint8_t channel)
+core::Result I2cMux<T, N>::selectChannel(const std::uint8_t channel)
 {
   if (channel >= kMaxNumChannels) {
-    logger_.log(core::LogLevel::kFatal, "Mux Channel number %d is not selectable", channel);
+    logger_.log(core::LogLevel::kFatal, "I2c Mux Channel number %d is not selectable", channel);
     return core::Result::kFailure;
   }
   const std::uint8_t channel_buffer = 1 << channel;
-  const auto i2c_write_result       = i2c_.writeByte(mux_address_, channel_buffer);
+  const auto i2c_write_result       = i2c_->writeByte(mux_address_, channel_buffer);
   if (i2c_write_result == core::Result::kSuccess) {
-    logger_.log(core::LogLevel::kInfo, "Mux Channel %d selected", channel);
+    logger_.log(core::LogLevel::kInfo, "I2c Mux Channel %d selected", channel);
     return core::Result::kSuccess;
   } else {
```

Comment:
don't need the `else`

Reply:
I don't think that removing this matters too much

Reply:
it is code that doesn't do anything so it is only visual noise to paraphrase something i think i saw franz say somewhere

fite me

Reply:
also should be `if result == failure return failure` then afterwards `return success`
in keeping with using guards as elsewhere

Reply:
fair will change that